samza-rest/src/main/java/org/apache/samza/monitor/JobsClient.java

```diff
@@ -79,7 +80,7 @@ public JobsClient(List<String> jobStatusServers) {
    */
   public JobStatus getJobStatus(JobInstance jobInstance) {
     Job job = retriableHttpGet(baseUrl -> String.format(ResourceConstants.GET_JOBS_URL, baseUrl,
-        jobInstance.getJobName(), jobInstance.getJobId()));
+        jobInstance.getJobName(), jobInstance.getJobId()), new TypeReference<Job>(){});
```

Comment:
This(`retriableHttpGet`) was shared by both `getJobStatus`, `getTasks` method within this class. 

Since the same logic is shared by both of these methods, it's kept separate to avoid duplication.

Reason for keeping it generic is both of these methods expect different type as result, passing in type for that.